[READY]Removal of circuits

Shdorsh opened the pull request:
[Changelogs]: # Removes the entirety of circuits.

:cl: Shdorsh
del: Circuitry module
/:cl:

[why]: # The deal goes like this: you folks lose circuitry, which you already hate, and hippie gets to modularize it and use it as they wish. Also, I'm still waiting for the map removal.

[EDIT] A little quote from the discord so you folks understand what's going on:

The main argument is that tg wanted circuits gone anyway, all the mantainers were clear about this. if I hadn't done it, someone else would, and they wouldn't have possibly tried to modularize it on another codebase and circuits would just be gone, finito. I'm trying to do all the stuff so I don't have to handle shitty merge conflicts, it's really the last thing I'd want since  I wouldn't be able to correct those.
If you want to, I can just close down all the prs and shit, no modularization at all and I won't remove circuits on tg. When the mantainers eventually do it as they planned though, your problem
you won't even have hippie to go and do some circuiting.
I know, for you folks, it must be like choosing between having their hand cut off or their arm and asking whether "neither" is a solution.

The circuit-less department:

https://cdn.discordapp.com/attachments/502557115475296276/504792653448085504/box_circuits_removal.PNG
https://cdn.discordapp.com/attachments/502557115475296276/504792650642358272/delta_circuits_removal.PNG
https://cdn.discordapp.com/attachments/502557115475296276/504792648385691657/pubby_circuits_removal.PNG
https://cdn.discordapp.com/attachments/502557115475296276/504792645105877012/meta_circuits_removal.PNG

Edit: The new circuit-less department: nano-style
https://cdn.discordapp.com/attachments/502557115475296276/506913460664139778/box_circuits_removal_2.0.PNG

Shdorsh committed:
Delete integrated_circuit.dm

Shdorsh committed:
Delete integrated_electronics.dm

Shdorsh committed:
Delete pins.dm

Shdorsh committed:
Delete wirer.dm

Shdorsh committed:
Delete helpers.dm

Shdorsh committed:
Delete printer.dm

Shdorsh committed:
Delete analyzer.dm

Shdorsh committed:
Delete debugger.dm

Shdorsh committed:
Delete detailer.dm

Shdorsh committed:
Delete assemblies.dm

Shdorsh committed:
Delete saved_circuits.dm

Shdorsh committed:
Delete boolean_pin.dm

Shdorsh committed:
Delete char_pin.dm

Shdorsh committed:
Delete color_pin.dm

Shdorsh committed:
Delete dir_pin.dm

Shdorsh committed:
Delete index_pin.dm

Shdorsh committed:
Delete list_pin.dm

Shdorsh committed:
Delete number_pin.dm

Shdorsh committed:
Delete ref_pin.dm

Shdorsh committed:
Delete selfref_pin.dm

Shdorsh committed:
Delete string_pin.dm

Shdorsh committed:
Delete passive.dm

Shdorsh committed:
Delete power.dm

Shdorsh committed:
Delete access.dm

Shdorsh committed:
Delete arithmetic.dm

Shdorsh committed:
Delete converters.dm

Shdorsh committed:
Delete data_transfer.dm

Shdorsh committed:
Delete input.dm

Shdorsh committed:
Delete lists.dm

Shdorsh committed:
Delete logic.dm

Shdorsh committed:
Delete manipulation.dm

Shdorsh committed:
Delete memory.dm

Shdorsh committed:
Delete output.dm

Shdorsh committed:
Delete power.dm

Shdorsh committed:
Delete reagents.dm

Shdorsh committed:
Delete smart.dm

Shdorsh committed:
Delete text.dm

Shdorsh committed:
Delete time.dm

Shdorsh committed:
Delete trig.dm

Shdorsh committed:
Update tgstation.dme

Shdorsh committed:
Update multitool.dm

Shdorsh committed:
Delete circuit.dm

Shdorsh committed:
Update tgstation.dme

Shdorsh committed:
Update netdata.dm

Shdorsh committed:
Update ntnet_interface.dm

Shdorsh committed:
stuff

Shdorsh committed:
Update packs.dm

Shdorsh committed:
Update electronics_designs.dm

Shdorsh committed:
Update electronics_designs.dm

Shdorsh committed:
Update multitool.dm

Shdorsh committed:
Update cavity_implant.dm

Shdorsh committed:
Update cards_ids.dm

Shdorsh committed:
Update walls.dm

Shdorsh committed:
Update hostile.dm

optimumtact commented:
just another example of why hippie can't code or do anything original

ExcessiveUseOfCobblestone commented:
> a webbit deleting an entire feature

really...

Garen7 commented:
He posted screen shots of himself compiling and working on it locally earlier
![image](https://user-images.githubusercontent.com/22408776/47403903-e4085500-d710-11e8-9516-4f4a0660af46.png)

Garen7 commented:
Also this is long over due,

Doing any work on circuits code has been difficult since the maintainers don't want to review it, not that they're bad people for not wanting to but it will still be much healthier somewhere where work on it is possible.

ShizCalev commented:
This'll require some remapping too my dude.

PKPenguin321 commented:
ho dont do it

granpawalton commented:
@ShizCalev I told him I could handle the remapping for him, ill be finished up tonight or tommarow

fen0s commented:
**[https://www.youtube.com/watch?v=umDr0mPuyQc](*NO! NO, GOD, PLEASE, NO! NOOOOOOOOOOOOOOO!*)

fen0s commented:
[**NO! NO, GOD, PLEASE, NO! NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO!**](https://www.youtube.com/watch?v=umDr0mPuyQc)

evsey9 commented:
lmao you could provide any other even SLIGHTLY bettrr reason which there are some to PR tgis but you choose the shittiest one
gj
if you want to modularize circuits so much then I guess you'll have to stop being an upstream

Shdorsh commented:
it's not about the server population not enjoying circuits, it's about the code maintainers taking a month to merge a

Shdorsh commented:
god i hate it when my ohone lags and makes me press the wrong button

Shdorsh commented:
Anyway, the problem isn't that circuits aren't well loved by the server population, but rather that it takes about one month to get a one line bugfix merged, with the possibility of it being closed down due to "low to none interest by code mantainers" and I get reprimanded just for grouping circuit PRs together like it's the end of the world.

Unless we elect a code mantainer who has an interest in those game mechanics or is entirely impartial on that stuff, circuits will become stale and die off anyway, I'm just doing y'all a favor by migrating it towards a codebase where it can grow further and we don't need months only to have the anti-eye cancer assembly UI possibly be closed because mantainers didn't like it.

Shdorsh commented:
@evsey9 I don't care if people hate this. I didn't come here to bring peace, but change, and through that, improvement.

evsey9 commented:
why exactly do you need to remove circuits here to modularize them on hippie

ExcessiveUseOfCobblestone commented:
> bundles changes
> complains about "the man" when asked to not bundle changes

Shdorsh commented:
It all fitted the circuitry QoL theme, if I remember right, so I thought that if it all is in the same category. It's not like I changed let's say changed how the tesla works, then added a new wiz spell and added a new research node in one PR.

Kawaii-Big-Boss commented:
I don't think even Hippiestation would want circuits around. Hippiestation players tend to want to break new features the moment they're added for the sake of powergaming, which resulted in us discovering that you can make maxcap bombs with circuits with roundstart, unchanged emergency oxygen tanks. No tritium or anything else like that was necessary. You could also make lightspeed drones that protect you from kidnapping or other shit similar to that (explosive drones are a favorite) which definitely did not help when people started seeing circuits as "powergaming that only coders will understand."
Scientist is already not really respected over here, ask anyone who has been on the server for a while and they'll likely say Scientist only exists for stupid, shitty powergame like xenobio or the newly added circuits, letting antisocial players sit around in their fort doing fuckall for the entire round until they manage to create something OP and unleashing it on everyone.

We had like seven PRs so far nerfing Circuits because of how abused they were over here. As a result, I don't think most people would notice or care if Circuits were removed. Saying you want to remove them so "hippiestation will modularize them" or whatever doesn't make much sense if you ask me.

Barhandar commented:
Remove them from downstream if you're abusing them; TG specifically has culture of not abusing circuits do they don't get yeeted out of the codebase.

subject217 commented:
the vast majority of circuit powergamers now play on hippie with the blatantly overpowered circuits. there are very, very few people left on /tg/ who use circuits, and there are very few of those who use circuits for anything remotely interesting or useful that isn't some tidbit of unnerfed powergaming. this is just rational.

Shdorsh committed:
Update all_nodes.dm

bgobandit commented:
>removing the only substantial new addition to non-antag gameplay in ages

optimumtact commented:
it might be substantial but it was overpowered, lacked logging and caused a lot of headaches for the admins, and the people we felt were making reasonable changes got harrassed by the circuit players and discord to the point where no maintainer was interested in looking at changes for it.

It failed to make it here I'm afraid, maybe in the future we can port it back from hippie

granpawalton committed:
removes circuits from maps

granpawalton committed:
Revert "removes circuits from maps"

This reverts commit 14daf7770220a8a1d7a8fa497c92d9c866939160.

granpawalton committed:
Removes circuits lab from all maps

granpawalton committed:
Merge branch 'patch-27' of https://github.com/Shdorsh/tgstation into patch-27

sabiram commented:
Are you doing this in the web editor?

Shdorsh commented:
@sabiram No, it's been tested before on private server, like nearly all of my stuff.

Shdorsh commented:
I work together with granpa, he does the mapping, I do the code.

granpawalton committed:
removes circuits lab from lavaland syndicate base

Euphausiacea commented:
![the end of circuits](https://user-images.githubusercontent.com/37985537/47467313-d19b2380-d7c3-11e8-9d3e-7f0a43331442.png)

Bawhoppen commented:
The replacement mapping on box is kinda questionable, IMO. Just make it like how it used to be before circuits.

nicbn commented:
Restore box science to how it looked before the circuit map change (RD to the north, RnD in a container room, experimentor at the end of the corridor etc.)

granpawalton commented:
@Bawhoppen on box circuits was originally added right here and got moved in a remapping of box done a while back

![image](https://user-images.githubusercontent.com/36310010/47470980-d9f85c00-d7cd-11e8-8efc-70f6daf28122.png)

Currently the RP room where you can gas people is in its place and circuitry is placed in between RD office, RnD, the main sci hallway and experimenter room so to remove it we can push one or some of those rooms in its place, or remap several rooms in sci

@nicbn do you have a picture of the time you're referencing or is it from before #39318

nicbn commented:
Deleted my comment because Bawhoppen comment synthesizes it.
Restore the north of science to how it looked before #39318. Take a look at how box looked before circuitry lab to have some inspiration on remapping that part. Maybe replace it with a firing range and extend the maintenance

BeeSting12 commented:
i still have ptsd from the beep bots. 👍

ExcessiveUseOfCobblestone commented:
> Remove them from downstream if you're abusing them; TG specifically has culture of not abusing circuits do they don't get yeeted out of the codebase.

You mean anymore. Had they not been abused we wouldn't have the issue of maintainers not wanting to touch the pr's with a 10 foot pole

ShizCalev commented on _maps/map_files/MetaStation/MetaStation.dmm:
```diff
@@ -83847,39 +83727,12 @@
-/turf/open/floor/plasteel,
-/area/science/circuit)
+/obj/effect/spawner/structure/window/reinforced,
+/turf/open/space/basic,
```
bad

Shdorsh replied:
huh, where dat from?

Shdorsh replied:
oh, that's a mapping issue for @granpawalton

ShizCalev commented on _maps/map_files/BoxStation/BoxStation.dmm:
```diff
@@ -56878,10 +56787,16 @@
+	name = "test chamber blast door"
+	},
+/obj/machinery/door/firedoor/heavy,
+/turf/open/space/basic,
```
bad

ShizCalev commented:
absolutely going to need pictures for these.

granpawalton committed:
fixes space tiles under windows on meta

granpawalton committed:
Deltastation fixes

granpawalton commented:
Ill handle the box fixes including the accidental change in the library with the changes requested by nicbn, which could take a bit because I need to catch up with some due dates

Theres been pictures in the PR description since I commited my changes or did you mean map diff bot

YakumoChen commented:
Why do we even have firing ranges again when science can't even print guns!

optimumtact commented:
it's to rub it in your face

81Denton commented:
Don't worry, now I can axe the firing ranges and put nanite labs there instead

81Denton commented on code/game/objects/items/devices/multitool.dm:
```diff
@@ -28,77 +28,12 @@
-	var/datum/integrated_io/selected_io = null  //functional for integrated circuits.
 	var/mode = 0
 
-/obj/item/multitool/examine(mob/user)
```
Keep this check and only remove the `if(selected_io)` lines; the buffer is used for stuff like linking tcomms and quantum pads

81Denton commented:
Circuitry needs to be removed from these as well:

https://github.com/tgstation/tgstation/blob/8981618ab8b480a91aa725fee2a1c93ec27b4653/code/_globalvars/lists/flavor_misc.dm#L145
https://github.com/tgstation/tgstation/blob/8981618ab8b480a91aa725fee2a1c93ec27b4653/code/_globalvars/lists/flavor_misc.dm#L166

kingoftheautism commented:
i ded, also see denton review above

Shdorsh committed:
Update multitool.dm

Shdorsh committed:
Update flavor_misc.dm

Shdorsh commented:
> 
> 
> Circuitry needs to be removed from these as well:
> 
> [tgstation/code/_globalvars/lists/flavor_misc.dm](https://github.com/tgstation/tgstation/blob/8981618ab8b480a91aa725fee2a1c93ec27b4653/code/_globalvars/lists/flavor_misc.dm#L145)
> 
> Line 145 in [8981618](/tgstation/tgstation/commit/8981618ab8b480a91aa725fee2a1c93ec27b4653)
>  24 Circuitry 
> 
> [tgstation/code/_globalvars/lists/flavor_misc.dm](https://github.com/tgstation/tgstation/blob/8981618ab8b480a91aa725fee2a1c93ec27b4653/code/_globalvars/lists/flavor_misc.dm#L166)
> 
> Line 166 in [8981618](/tgstation/tgstation/commit/8981618ab8b480a91aa725fee2a1c93ec27b4653)
>  "Circuitry", "Toxins", "Dormitories", "Virology",

Done

nicbn commented:
Another way of redoing the box map is replacing it with nanite lab

Ratvar commented:
Will there be R&D cost rebalance due to circuitry research loss?

Shdorsh commented:
Huh, where's the merge conflict coming from, @Cyberboss ?

Shdorsh commented:
Where the hell did that come from? Didn't we say we wanted to remove circuitry? Why was it accepted?

evsey9 commented:
Who is "we"?

Cyberboss commented:
Maintainers

On Tue, Oct 30, 2018, 8:40 AM Evsey Antonovich <notifications@github.com>
wrote:

> Who is "we"?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/41108#issuecomment-434284716>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHyweugdppYCNqKiHMU1boWB01eO6dHCks5uqEjWgaJpZM4X2_aN>
> .
>

granpawalton committed:
Replaces Box circuits with nanites

granpawalton committed:
probably fixes merge conflict

granpawalton commented:
Everything should be good to go, idk why map bots mad

I didnt even realize box didnt have nanites at first so good call lol

Shdorsh commented:
Ok, now's the time to modularize.

kingoftheautism commented:
we didnt got flight suits back either

On 1 Nov 2018 08:23, "GunHog" <notifications@github.com> wrote:

> We are never getting circuits back now :(
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/41108#issuecomment-434881027>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AmZ1u0mjWO1vwH7sYsjr28ov0WBHW6VIks5uqjEIgaJpZM4X2_aN>
> .
>

PKPenguin321 commented:
I hurt myself today
To see if I still feel
I focus on the pain
The only thing that's real
The needle tears a hole
The old familiar sting
Try to kill it all away
But I remember everything
What have I become
My sweetest friend
Everyone I know
Goes away in the end
And you could have it all
My empire of dirt
I will let you down
I will make you hurt
I wear this crown of thorns
Upon my liars chair
Full of broken thoughts
I cannot repair
Beneath the stains of time
The feelings disappear
You are someone else
I am still right here
What have I become
My sweetest friend
Everyone I know
Goes away in the end
And you could have it all
My empire of dirt
I will let you down
I will make you hurt
If I could start again
A million miles away
I will keep myself
I would find a way

BootlegBow commented:
am sad

Cyberboss commented:
Don't necropost
